### ISSUE TITLE
extraParams.hxml not importing libs unless dummy -cp

### ISSUE THREAD

**Issue by Dvergar** [HaxeFoundation/haxe#4059]
It seems that extraParams.hxml won't import any library given by `-lib mylib`.

@nadako investigated this a bit for me so i will just throw a few quotes from IRC

> it does process arguments after adding libs, but it will only call process_libs again on -cp
> that could be fixed by adding a loop inside lib processing in haxe

The issue is indeed solved with a dummy -cp like

```
-lib mylib
-cp "dummy"
```

**Comment by fzzr-** [comment]
extraParams.hxml is only for initial macro `--macro` and defines `-D `.

**Comment by ibilon** [comment]
In that specific case you could just add the library in the haxelib.json,
but personally I think every valid hxml should work in extraParams.

Also isn't that an haxelib issue?

**Comment by fzzr-** [comment]
>Also isn't that an haxelib issue?

No. Haxelib client prints full content of extraParams.hxml ([code](https://github.com/HaxeFoundation/haxelib/blob/f8bb48ba5a59c0dbab7a3373cdc908e9c1678644/src/haxelib/client/Main.hx#L1231-L1232)).

>but personally I think every valid hxml should work in extraParams

I think it's not because there no need any other args like about targets or `--no-output`, `--no-inline` or `-lib`, `-cp` - personally I think it will be look like shooting yourself in the foot, imho.

**Comment by Simn** [comment]
I have no idea what extraParams.hxml is and isn't supposed to support.

**Comment by ncannasse** [comment]
extraParams.hxml is an haxelib feature: if you add it to your library,
"haxelib path library" will return these parameters additionally to path
and -D

**Comment by filt3rek** [comment]
Hej there !
I encourred the same issue yesterday and openend another issue, because I haven't seen this one.
_haxelib.json_, permits to add dependecies with the `dependencies` param.
And I thought we could put any build file param in _extraParams.hxml_ 
So I'm a bit lost at the end : To add a dependecie I use _haxelib.json_ OK.
But how can I force adding another lib in the project (the dependecy), someone who add mylib in his project please ?

**Comment by kLabz** [comment]
Adding it as a dependency in haxelib.json already does add the lib to the project

**Comment by filt3rek** [comment]
Hej Rudy !
Thx for your quick answer !
Waw, didn't knew that, I've tested and it works, you're right !
Have a nice day !

**Comment by filt3rek** [comment]
...but BTW it would be cool when we have the time of course, it's not a priority I think, that we clarify what is permit in the file _extraParams.hxml_ because instinctively, for me, we could write any valid haxe build param...

**Comment by Simn** [comment]
This should work properly now.
